From 7ad110db505c280201edb22939b777f83fcf52a7 Mon Sep 17 00:00:00 2001
From: Sebastiano Barezzi <seba@sebaubuntu.dev>
Date: Wed, 11 Oct 2023 15:53:20 +0200
Subject: [PATCH 068/115] drivers: input: touchscreen: xiaomi:
 goodix_driver_gt9886: Give fod_status ops to tp_common

Change-Id: I785bfd8cb1c89742c55bdb1f82933289545214ed
---
 .../goodix_driver_gt9886/goodix_ts_core.c     | 44 +++++++++----------
 1 file changed, 21 insertions(+), 23 deletions(-)

diff --git a/drivers/input/touchscreen/xiaomi/goodix_driver_gt9886/goodix_ts_core.c b/drivers/input/touchscreen/xiaomi/goodix_driver_gt9886/goodix_ts_core.c
index 6e5a5ec29..b8c7ddb5c 100644
--- a/drivers/input/touchscreen/xiaomi/goodix_driver_gt9886/goodix_ts_core.c
+++ b/drivers/input/touchscreen/xiaomi/goodix_driver_gt9886/goodix_ts_core.c
@@ -1332,32 +1332,31 @@ static ssize_t gtp_fod_test_store(struct device *dev,
 	return count;
 }
 
-static ssize_t gtp_fod_status_show(struct device *dev,
-				   struct device_attribute *attr, char *buf)
+#ifdef CONFIG_TOUCHSCREEN_COMMON
+static ssize_t gtp_fod_status_show(struct kobject *kobj,
+				   struct kobj_attribute *attr, char *buf)
 {
-	struct goodix_ts_core *core_data = dev_get_drvdata(dev);
-
-	return snprintf(buf, 10, "%d\n", core_data->fod_status);
+	return snprintf(buf, 10, "%d\n", goodix_core_data->fod_status);
 }
 
-static ssize_t gtp_fod_status_store(struct device *dev,
-				    struct device_attribute *attr,
+static ssize_t gtp_fod_status_store(struct kobject *kobj,
+				    struct kobj_attribute *attr,
 				    const char *buf, size_t count)
 {
-	struct goodix_ts_core *core_data = dev_get_drvdata(dev);
-	//struct goodix_ts_event *ts_event = &core_data->ts_event;
-	ts_info("buf:%s, count:%zu\n", buf, count);
-	sscanf(buf, "%u", &core_data->fod_status);
+	sscanf(buf, "%u", &goodix_core_data->fod_status);
 
-	//goodix_ts_input_report(core_data->input_dev,&ts_event->event_data.touch_data);
-	core_data->gesture_enabled = core_data->double_wakeup |
-				     core_data->fod_status;
-	goodix_check_gesture_stat(!!core_data->fod_status);
+	goodix_core_data->gesture_enabled = goodix_core_data->double_wakeup |
+					    goodix_core_data->fod_status;
+	goodix_check_gesture_stat(!!goodix_core_data->fod_status);
 
 	return count;
 }
-static DEVICE_ATTR(fod_status, (S_IRUGO | S_IWUSR | S_IWGRP),
-		   gtp_fod_status_show, gtp_fod_status_store);
+
+static struct tp_common_ops fod_status_ops = {
+	.show = gtp_fod_status_show,
+	.store = gtp_fod_status_store,
+};
+#endif
 
 static DEVICE_ATTR(fod_test, (S_IRUGO | S_IWUSR | S_IWGRP), NULL,
 		   gtp_fod_test_store);
@@ -2765,6 +2764,11 @@ static int goodix_ts_probe(struct platform_device *pdev)
 	if (r < 0) {
 		ts_err("Failed to create double_tap node err=%d\n", r);
 	}
+
+	r = tp_common_set_fod_status_ops(&fod_status_ops);
+	if (r < 0) {
+		ts_err("Failed to create fod_status node err=%d\n", r);
+	}
 #endif
 
 	r = ts_device->hw_ops->reset(ts_device);
@@ -2849,12 +2853,6 @@ static int goodix_ts_probe(struct platform_device *pdev)
 		goto out;
 	}
 
-	if (sysfs_create_file(&core_data->gtp_touch_dev->kobj,
-			      &dev_attr_fod_status.attr)) {
-		ts_err("Failed to create fod_status sysfs group!\n");
-		goto out;
-	}
-
 	if (sysfs_create_file(&core_data->gtp_touch_dev->kobj,
 			      &dev_attr_fod_test.attr)) {
 		ts_err("Failed to create fod_test sysfs group!");
-- 
2.52.0

